### ISSUE TITLE
Formulas ignore NA

### ISSUE THREAD

**Issue by CSHornby** [PHPOffice/PhpSpreadsheet#2066]
This is:

```
- [X ] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
e.g. INDEX uses the result of MATCH to lookup another value. If MATCH returns #N/A then INDEX also returns #N/A. This is Excel behaviour.

### What is the current behavior?
INDEX treats #N/A as 0 and returns the first item in the array

### What are the steps to reproduce?

Please provide a [Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve) of code that exhibits the issue without relying on an external Excel file or a web server:

The following code uses the spreadsheet in the attached tar.gz: 
[test.xlsx.tar.gz](https://github.com/PHPOffice/PhpSpreadsheet/files/6427943/test.xlsx.tar.gz)

It should print:
#N/A
#N/A
Error

but instead prints:
#N/A
a
a

```php
<?php

use PhpOffice\PhpSpreadsheet\Reader\Xlsx;

require_once __DIR__ . '/vendor/autoload.php';

$reader = new Xlsx();

$book = $reader->load('test.xlsx');

$sheet = $book->getSheet(0);

print($sheet->getCell('G3')->getCalculatedValue()) . PHP_EOL;
print($sheet->getCell('E3')->getCalculatedValue()) . PHP_EOL;
print($sheet->getCell('C3')->getCalculatedValue()) . PHP_EOL;

// add code that show the issue here...
```

### Which versions of PhpSpreadsheet and PHP are affected?
Versions tried (but probably more)

PhpSpreadsheet  1.11 -> 1.17
PHP 7.2 -> 7.4

It looks like PhpOffice\PhpSpreadsheet\Calculation\Calculation::ProcessTokenStack doesnt attemp to handle #N/A at all, where's it could spot #N/As in $argArrayVals and set the $result as #N/A

**Comment by MarkBaker** [comment]
This isn't something that should be addressed in processing the token stack, because there are valid cases where errors can be passed as arguments to function implementations (e.g. `IFNA()`, and functions like COUNT() and its variations have rules that determine how error values should be handled). Instead, it should be handled by the function implementation itself to determine how an error should be processed.

I'll take a look at the implementation of INDEX()

**Comment by CSHornby** [comment]
OK - I think the same issue affects a number of functions (VLOOKUP, HLOOKUP, LOOKUP etc)

**Comment by MarkBaker** [comment]
That's certainly possible; many of them were originally coded for happy path when they were first implemented; and while we've recently done some intensive refactoring (now merged into the master branch) and improved the testing for many of these implementations, it's still a work in progress. I'll check against the other Lookup and Reference functions while I'm looking at INDEX()

**Comment by oleibman** [comment]
Fixed by PR #2066 and/or PR #2939. Closing.
